### PR TITLE
chore: instrumentation added for generate crud create action

### LINKED ISSUE #31171: [Task]: Add accelerator and isUserCreated properties in create_NEWACTION event for generate crud
### SubTasks

Notion doc: https://www.notion.so/appsmith/Add-accelerator-and-isUserCreated-properties-in-create_NEWACTION-event-for-generate-crud-5f93060be9804d559eb691c79305bc1d

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR adds `create_NEWACTION` instrumentation for generate crud flow. Today generate crud flow is handled by `/crud` API, this API is responsible for creating actions, widgets and everything that is needed for generating crud app. When actions are created in generate crud flow, `create_NEWACTION` event is triggered, we wanted to add details like `isUserCreated` and `accelerator` to this event, which will help us in segregation of actions based on whether they were created manually or by generate crud accelerator. 

In this PR, we are passing the event context of generate crud when calling create action function, this context is then eventually added to the event properties.

**Test cases:**
- When generate crud flow is triggered, `create_NEWACTION` event should have two more props `isUserCreated:  false` and `accelerator: "generate-crud`
- When actions are created by any other method, these two properties wont be there.

#### PR fixes following issue(s)
Fixes #31171 
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced a new event type `GENERATE_PAGE` for enhanced analytics and event tracking capabilities.
- **Refactor**
	- Modified analytics properties generation across various services to include context-specific information, improving the accuracy and relevance of analytics data collected.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/base/ActionCollectionServiceCEImpl.java [hunk 1]
```diff
@@ -697,7 +697,7 @@ protected Mono<ActionDTO> createJsAction(ActionCollection actionCollection, Acti
         newActionService.updateDefaultResourcesInAction(newAction);
 
         Mono<NewAction> sendAnalyticsMono =
-                analyticsService.sendCreateEvent(newAction, newActionService.getAnalyticsProperties(newAction));
+                analyticsService.sendCreateEvent(newAction, newActionService.getAnalyticsProperties(newAction, null));
```
[the remark this answers is not in the document]
Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/base/ActionCollectionServiceCEImpl.java:700]: @sneha122 just to confirm, have you made sure that `createJsAction()` is not triggered from any of the non-user generated flows?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java [hunk 2]
```diff
@@ -1634,6 +1637,12 @@ public Map<String, Object> getAnalyticsProperties(NewAction savedAction) {
                     "dsIsMock",
                     ObjectUtils.defaultIfNull(unpublishedAction.getDatasource().getIsMock(), ""));
         }
+
+        if (eventContext != null
+                && AppsmithEventContextType.GENERATE_PAGE.equals(eventContext.getAppsmithEventContextType())) {
+            analyticsProperties.put("isUserCreated", false);
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java:1643]: Hey so I actually thought of doing this, but the thing is AppsmithEventContextType has two other values called "DEFAULT" and "CLONE_PAGE", For default one, it makes sense to keep isUserCreated as true but I am not sure about clone page, let me know what you think, I can then add else condition here

File: app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/base/ActionCollectionServiceCEImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sneha122) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/base/ActionCollectionServiceCEImpl.java:700]: @NilanshBansal It is definitely not triggered from generate crud flow, hence I have added it as null

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (NilanshBansal) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java:1643]: I believe whatever we are sure of we should add. 
So, for the DEFAULT type we can add the userCreated as true and leave the other one as it is for now. As and when we pick up the other flow we should populate these values later.

[the remark this answers is not in the document]
Comment (sneha122) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java:1643]: Makes sense, I will update the code

**Review by NilanshBansal — APPROVED** [review]

**Pull request merged**